src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -55,23 +57,25 @@ internal static string NormalizeDirectorySeparators(string path)
                 }
             }
 
-            if (normalized) return path;
+            if (normalized)
+                return path;
 
-            StringBuilder builder = new StringBuilder(path.Length);
-
-            for (int i = 0; i < path.Length; i++)
+            return string.Create(path.Length, path, (dst, state) =>
             {
-                char current = path[i];
-
-                // Skip if we have another separator following
-                if (IsDirectorySeparator(current)
-                    && (i + 1 < path.Length && IsDirectorySeparator(path[i + 1])))
-                    continue;
+                int j = 0;
+                
+                for (int i = 0; i < path.Length; i++)
+                {
+                    char current = path[i];
 
-                builder.Append(current);
-            }
+                    // Skip if we have another separator following
+                    if (IsDirectorySeparator(current)
+                        && (i + 1 < path.Length && IsDirectorySeparator(path[i + 1])))
+                        continue;
 
-            return builder.ToString();
+                    dst[j++] = current;
```

Comment:
This may not be filling all the characters that you have asked for in string.Create. The string can have extra zeros at the end if more than we have skipped the separators more than once.

Reply:
Can we revert this method and deal with optimizing it separately? It isn't terrible as is and as such I'd prefer to keep the enhancement a discrete change.

Reply:
I will revert it use stringbuilder